### PR TITLE
TokenProvider to handle auth logic

### PR DESCRIPTION
**Pull request by ldanilina**
- Auth logic (both internal and external) has been extracted into TokenProvider
- CI pipeline to propagate internal values will be added as a separate MR

### CODE REVIEW

File: packages/mobile/src/providers/TokenUrl.tsx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const URL_TOKEN: string | null = null;
```
Comment (aravind-raveendran) [inline packages/mobile/src/providers/TokenUrl.tsx:1]: Would this be the file that we write the actual urls to?

Comment (ldanilina) [reply packages/mobile/src/providers/TokenUrl.tsx:1]: Yes. will provide details

**Review by aravind-raveendran — APPROVED** [review]

File: packages/mobile/src/providers/TokenProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+type TokenContext = {
+  token: string | null;
+  setToken: (value: string | null) => void;
+  refreshToken: (() => Promise<string>) | null;
+};
+
+type TokenProviderProps = {
+  children: ReactElement;
+  validateToken: (token: string) => boolean;
+  fetch: any;
```
Comment (CasperMok) [inline packages/mobile/src/providers/TokenProvider.tsx:19]: Get some warning for `any`. Is `any` a most?
If so, can we disable the warning?

Comment (ldanilina) [reply packages/mobile/src/providers/TokenProvider.tsx]: updated the type to be more specific

**Review by aravind-raveendran — APPROVED** [review]

**Review by CasperMok — APPROVED** [review]

**Pull request merged**